### ISSUE TITLE
[Bug]: Decompressing and inserting into a compressed chunk produces incorrect catalog data

### ISSUE THREAD

**Issue by antekresic** [timescale/timescaledb#5419]
### What type of bug is this?

Incorrect result

### What subsystems and features are affected?

Compression

### What happened?

While simultaneously decompressing and inserting into a compressed chunk, resulting chunk can end up with an invalid chunk status (8) which basically means the chunk is uncompressed and partially compressed at the same time.

### TimescaleDB version affected

main branch

### PostgreSQL version used

14.5

### What operating system did you use?

Ubuntu 22.04.1

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
2023-03-08 14:28:30.953 CET  [267298]  LOG:  00000: server process (PID 267392) was terminated by signal 6: Aborted     
2023-03-08 14:28:30.953 CET  [267298]  DETAIL:  Failed process was running: BEGIN; INSERT INTO ts_device_table VALUES (1, 1, 100, 100);
```

### How can we reproduce the bug?

```bash
Block chunk status on separate session:
;
  SELECT status as chunk_status from _timescaledb_catalog.chunk
  WHERE id = ( select min(ch.id) FROM _timescaledb_catalog.hypertable ht, _timescaledb_catalog.chunk ch WHERE ch.hypertable_id = ht.id AND ht.table_name like 'ts_device_table') FOR UPDATE;

Start decompress of chunk on different session:
 BEGIN;
  SET LOCAL client_min_messages TO WARNING;
  SELECT
    CASE WHEN decompress_chunk(ch, true) IS NOT NULL THEN true ELSE false END AS decompress
  FROM show_chunks('ts_device_table') AS ch
  ORDER BY ch::text LIMIT 1;
```
Start insertion on third session:
``` BEGIN; INSERT INTO ts_device_table VALUES (1, 1, 100, 100); ```
Unblock chunk status on first session which unblocks decompression:
``` ROLLBACK;```
Commit decompression on second session and issue should occur:
``COMMIT;```

What you should see is an assertion fail for debug builds or chunk status 8 for the chunk in question.
```

**Comment by konskov** [comment]
closing as duplicate of #5389
